serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -31,6 +34,49 @@
 LOG = structlog.stdlib.get_logger(__name__)
 
 
+def fetch_raw_logprob_infos(
+    logits,
+    res_tokens,
+    sampling_params,
+) -> List[Optional[RawLogprobsInfo]]:
+    logprob_infos: List[Optional[RawLogprobsInfo]] = []
+    num_seq = logits.shape[0]
+    for index in range(num_seq):
```

Comment:
I think we can run this check on CPU async and vectorize the computation further as in https://github.com/octoml/mlc-llm/pull/148. Let me follow-up once this is merged.